### PR TITLE
Test driven development

### PR DESCRIPTION
**Pull request by FilbertKimasa**
**In this PR we implemented the following:**

- [ ] Create tests for Solver, factorial, reverse and fizzbuzz  functions
- [ ] Create Solver class with Solver, factorial, reverse, and fizzbuzz  functions
- [ ] Run the tests

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+require_relative '../tdd/solver'
+
+describe Solver do
+  describe '#factorial' do
+    it 'returns 1 for 0' do
+      solver = Solver.new
+      expect(solver.factorial(0)).to eq(1)
+    end
+  end
```
Comment (tresorsawasawa) [inline spec/solver_spec.rb]: - [ ] As required for the project, kindly ensure that the `#factorial` method also has test for **Negative value which raises an exception as an expectation** and **Special case - when input is 0, the result should equal 1**

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @FilbertKimasa and @NoelLincoln 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Follow TDD approach :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**I reviewed this project with[ advice of the Sr. Reviewer](https://microverse-crs.slack.com/archives/CM454ATKJ/p1702129009349689)**

**Review by DeliceLydia — APPROVED** [review]
## Hi @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
